### PR TITLE
DOC: Documentation for JuiceFS on Kubernetes

### PR DESCRIPTION
**Pull request by Zhou1213CN**
## Related issue number

Fixes #xxxx

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass

### CODE REVIEW

File: doc/source/user_guide/external_storage.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.. _external_storage_index:
```
Comment (ChengjieLi28) [inline doc/source/user_guide/external_storage.rst:1]: I think this page should be placed under ``Deployment -> Kubernetes deployment``, and also mentioned in ``Cloud deployment``.

File: doc/source/user_guide/deployment_juicefs_on_kubernetes.rst [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+    redis   1/1     Running   0           6d6h
+
+..
+
+
+Check redis pod's IP address, cpu limits and requests. In this example, IP for redis is 172.17.0.8.
+
+.. code-block:: bash
+
+    $ kubectl describe po redis
```
Comment (ChengjieLi28) [inline doc/source/user_guide/deployment_juicefs_on_kubernetes.rst:41]: Here you just want to verify redis' IP. You can restrict the kubectl output.Now this takes up too much space  on the page.

File: doc/source/user_guide/deployment_juicefs_on_kubernetes.rst [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+
+..
+
+
+JuiceFS Installation
+----------
+
+You can skip this installation part and jump directly to ``Deploy Cluster`` section.
+
+With the ``new_cluster`` function we provide in the next ``Deploy Cluster`` section, you can use JuiceFS directly without any installation.
```
Comment (ChengjieLi28) [inline doc/source/user_guide/deployment_juicefs_on_kubernetes.rst:134]: ``new_cluster`` function just install the PV and PVC for xorbits namespace, right? So User must install juicefs csi on K8s.

File: doc/source/user_guide/deployment_juicefs_on_kubernetes.rst [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+
+But here we will still walk you through the process of installing JuiceFS on a Kubernetes cluster, enabling you to leverage its features and benefits.
+
+You can choose one of the three ways on `Use JuiceFS on Kubernetes <https://juicefs.com/docs/zh/community/how_to_use_on_kubernetes>`_.
+
+Here we use CSI driver as an example.
```
Comment (ChengjieLi28) [inline doc/source/user_guide/deployment_juicefs_on_kubernetes.rst:140]: Our impl way in k8s must rely on CSI. So here you should clearly articulate this point.

File: doc/source/user_guide/deployment_kubernetes.rst [hunk 5]
```diff
@@ -95,3 +95,407 @@ Refer `DockerFile <https://github.com/xprobe-inc/xorbits/blob/main/python/xorbit
...
+
+.. code-block:: bash
+
+  resources:
+    limits:
+      cpu: 100m
+      memory: 50Mi
+    requests:
+      cpu: 100m
+      memory: 50Mi
```
Comment (ChengjieLi28) [inline doc/source/user_guide/deployment_kubernetes.rst:217]: I think what is given here is a minimal configuration that needs to be clarified in the text.

File: doc/source/user_guide/deployment_kubernetes.rst [hunk 6]
```diff
@@ -95,3 +95,407 @@ Refer `DockerFile <https://github.com/xprobe-inc/xorbits/blob/main/python/xorbit
...
+
+..
+
+
+3. Execute below commands to deploy JuiceFS CSI Driver:
+
+.. code-block:: bash
+
+    $ helm repo add juicefs https://juicedata.github.io/charts/
+    $ helm repo update
```
Comment (ChengjieLi28) [inline doc/source/user_guide/deployment_kubernetes.rst:227]: These two lines have already existed above.

File: doc/source/user_guide/deployment_kubernetes.rst [hunk 7]
```diff
@@ -95,3 +95,407 @@ Refer `DockerFile <https://github.com/xprobe-inc/xorbits/blob/main/python/xorbit
...
+    NAME                       READY   STATUS    RESTARTS   AGE
+    juicefs-csi-controller-0   3/3     Running   0          22m
+    juicefs-csi-node-v9tzb     3/3     Running   0          14m
+
+..
+
+Create and use PV
+^^^^^^^^^^^^^^^^^^^^^^
+
+You can skip this ``Create and use PV`` section because in Xorbits, the ``new_cluster`` function would create secret, pv, and pvc for you.
```
Comment (ChengjieLi28) [inline doc/source/user_guide/deployment_kubernetes.rst:246]: This sentence can be bolded.

File: doc/source/user_guide/deployment_kubernetes.rst [hunk 8]
```diff
@@ -95,3 +95,407 @@ Refer `DockerFile <https://github.com/xprobe-inc/xorbits/blob/main/python/xorbit
...
+
+You can still walk through this section as it would give you a better understanding of each parameter in the configurations.
+
+JuiceFS leverages persistent volumes to store data.
+
+Reference Page: `Create and use pv <https://juicefs.com/docs/csi/guide/pv>`_
+
+We would create several YAML files. Validate their formats on `YAML validator <https://www.yamllint.com/>` before usage.
```
Comment (ChengjieLi28) [inline doc/source/user_guide/deployment_kubernetes.rst:254]: image
Style issue here.

File: doc/source/user_guide/deployment_kubernetes.rst [hunk 9]
```diff
@@ -95,3 +95,407 @@ Refer `DockerFile <https://github.com/xprobe-inc/xorbits/blob/main/python/xorbit
...
+    default                                       Active   38d
+    kube-node-lease                               Active   38d
+    kube-public                                   Active   38d
+    kube-system                                   Active   38d
+    xorbits-ns-cc53e351744f4394b20180a0dafd8b91   Active   4m5s
+
+    $ kubectl get po -n xorbits-ns-cc53e351744f4394b20180a0dafd8b91
+    NAME                                 READY   STATUS             RESTARTS   AGE
+    xorbitssupervisor-84754bf5f4-dcstd   0/1     Running            0          80s
+    xorbitsworker-5b9b976767-sfpkk       0/1     Running            0          80s
```
Comment (ChengjieLi28) [inline doc/source/user_guide/deployment_kubernetes.rst:469]: You should provive a valid output here. The output here should be ``1/1``.

**Review by aresnow1 — APPROVED** [review]
LGTM

**Review by ChengjieLi28 — APPROVED** [review]

**Pull request merged**